extensions/wikia/CuratedContent/CuratedContentValidator.class.php

```diff
@@ -14,188 +15,170 @@ class CuratedContentValidator {
-	}
-
-	public function reset() {
-		$this->errors = [ ];
-		$this->existingSectionLabels = [ ];
-		$this->existingItemLabels = [ ];
-		$this->existingFeaturedItemLabels = [ ];
-		$this->hasOptionalSection = false;
-	}
-
 	public function validateData( $data ) {
-		$this->reset();
-
-		// validate sections
-		foreach ( $data as $section ) {
-			if ( !empty( $section['featured'] ) ) {
-				$this->validateItems( $section, true );
-			} else {
-				$this->validateSection( $section );
-				$this->validateItemsExist( $section );
-				$this->validateItems( $section );
-				$this->validateItemsTypes( $section );
+		$errors = [ ];
+		$alreadyUsedSectionLabels = [ ];
+
+		if ( is_array( $data ) ) {
+			foreach ( $data as $section ) {
+
+				if ( is_array( $section['items'] ) ) {
+					if ( !empty( $section['featured'] ) ) {
+						foreach ( $section['items'] as $featuredItem ) {
+							if ( !empty( $this->validateFeaturedItem( $featuredItem ) ) ) {
+								$errors[] = self::ERR_OTHER_ERROR;
+							} ;
+						}
+					} else {
+						$alreadyUsedSectionLabels[] = $section['title'];
+
+						if ( empty( $section['title'] ) ) {
+							foreach ( $section['items'] as $item ) {
+								if ( !empty( $this->validateSectionItem( $item ) ) ) {
+									$errors[] = self::ERR_OTHER_ERROR;
+								} ;
+							}
+						} else {
+							if ( !empty( $this->validateSectionWithItems( $section ) ) ) {
+								$errors[] = self::ERR_OTHER_ERROR;
+							} ;
+						}
+					}
+				}
+			}
+
+			foreach ( array_count_values( $alreadyUsedSectionLabels ) as $label => $count ) {
+				if ( $count > 1 ) {
+					$errors[] = self::ERR_DUPLICATED_LABEL;
+				}
 			}
 		}
-		// also check for duplicate labels
-		$this->validateDuplicatedLabels();
 
-		return $this->errors;
+		return $errors;
 	}
 
-	public function validateDuplicatedLabels() {
-		foreach ( array_count_values( $this->existingFeaturedItemLabels ) as $label => $count ) {
-			if ( $count > 1 ) {
-				$this->error( $label, 'featured', self::ERR_DUPLICATED_LABEL );
-			}
-		}
-		foreach ( array_count_values( $this->existingSectionLabels ) as $label => $count ) {
-			if ( $count > 1 ) {
-				$this->error( $label, 'section', self::ERR_DUPLICATED_LABEL );
-			}
-		}
-		foreach ( array_count_values( $this->existingItemLabels ) as $label => $count ) {
-			if ( $count > 1 ) {
-				$this->error( $label, 'item', self::ERR_DUPLICATED_LABEL );
-			}
-		}
+	private static function needsArticleId( $type ) {
+		return !in_array( $type, [ CuratedContentHelper::STR_CATEGORY ] );
 	}
 
-	public function getErrors() {
-		return $this->errors;
+	private static function isSupportedProvider( $provider ) {
+		return ( $provider === 'youtube' ) || ( startsWith( $provider, 'ooyala' ) );
 	}
 
-	private function error( $labelOrTitle, $type, $errorString ) {
-		if ( !empty( $errorString ) ) {
-			$this->errors[] = [ 'target' => $labelOrTitle, 'type' => $type, 'reason' => $errorString ];
+	public function validateFeaturedItem( $item ) {
```

Comment:
in this case we would need to pass errors array as a reference.
I would prefer to leave it that way because it increases readability